stylelint-polaris/plugins/custom-properties-allowed-list/index.test.js

```diff
@@ -0,0 +1,171 @@
+const internalPolarisComponentCustomProperties = /--pc-.+/;
+const internalConfig = [
+  {
+    allowedProperties: [
+      '--polaris-version-number',
+      internalPolarisComponentCustomProperties,
+    ],
+    allowedValues: {
+      '/.+/': [
+        internalPolarisComponentCustomProperties,
+        ...polarisCustomPropertyNames,
+      ],
+    },
+  },
+];
+
+testRule({
+  ruleName,
+  plugins: [__dirname],
+  config: externalConfig,
+  accept: [
+    {
+      code: '.a { border: 1px red; }',
+      description: 'Uses no custom properties',
+    },
+    {
+      code: '.a { --test: red; }',
+      description:
+        "Defining custom-properties that don't start with --p- or --pc-",
+    },
+    {
+      code: '.a { color: --p-text; }',
+      description: 'Using custom-properties from polaris-tokens',
+    },
+    {
+      code: '.a { --test: red; color: --test; }',
+      description: 'Using other custom-properties',
+    },
+  ],
+
+  reject: [
+    {
+      code: '.a { --p-test: red; }',
+      description:
+        'Defining custom-properties that start with --p- is disallowed',
+      message: messages.rejected('--p-test'),
+      line: 1,
+      column: 6,
+      endLine: 1,
+      endColumn: 20,
+    },
+    {
+      code: '.a { --pc-test: red; }',
+      description:
+        'Defining custom-properties that start with --pc- is disallowed',
+      message: messages.rejected('--pc-test'),
+      line: 1,
+      column: 6,
+      endLine: 1,
+      endColumn: 21,
+    },
+    {
+      code: '.a { color: --p-absent; }',
+      description:
+        'Using --p- prefixed tokens that do not exist in polaris-tokens is disallowed',
+      message: messages.rejected(undefined, '--p-absent'),
+      line: 1,
+      column: 6,
+      endLine: 1,
+      endColumn: 24,
+    },
+    {
+      code: '.a { color: --pc-test; }',
+      description: 'Using --pc- prefixed tokens is disallowed',
+      message: messages.rejected(undefined, '--pc-test'),
+      line: 1,
+      column: 6,
+      endLine: 1,
+      endColumn: 23,
+    },
+  ],
+});
+
+testRule({
+  ruleName,
+  plugins: [__dirname],
+  config: internalConfig,
+  accept: [
+    {
+      code: '.a { --pc-test: red; }',
+      description: 'Defining custom-properties that start with --pc-',
+    },
+    {
+      code: '.a { color: --p-text; }',
+      description: 'Using custom-properties from polaris-tokens',
+    },
+    {
+      code: '.a { color: --pc-my-value; }',
+      description: 'Using custom-properties that start with --pc-',
+    },
+  ],
+  reject: [
+    {
+      code: '.a { --p-test: red; }',
+      description:
+        'Defining custom-properties that start with --p- is disallowed',
+      message: messages.rejected('--p-test'),
+      line: 1,
+      column: 6,
+      endLine: 1,
+      endColumn: 20,
+    },
+    {
+      code: '.a { --test: red; }',
+      description:
+        "Defining custom-properties that don't start with --p- or --pc- is disallowed",
```

Comment:
```suggestion
        "Defining custom-properties that don't start with --pc- is disallowed",
```

Removing `--p-` as those should all be generated

Reply:
This block is for the internal config. Defining prpoerties that begin with `--p` in css is disallowed (see the test directly above this one). the `--p` properties are all provided by the CustomProperties component, so you're not allowed to create them in css files.

Reply:
> the `--p` properties are all provided by the CustomProperties component, so you're not allowed to create them in css files.

Right 👍 However, this `description` basically states you ARE allowed to define `--p-` custom properties.
e.g. the contrapositive
```
Defining custom-properties that start with --p- or --pc- are allowed
```

Unless I'm misunderstanding (double negatives are hard) and thus I apologies. Feel free to resolve the issue.